pytorch_lightning/loops/fit_loop.py

```diff
@@ -0,0 +1,254 @@
+
+    @current_epoch.setter
+    def current_epoch(self, value: int):
+        self.iteration_count = value
+
+    @property
+    def global_step(self):
+        return self.training_loop.global_step
+
+    @global_step.setter
+    def global_step(self, value):
+        self.training_loop.global_step = value
+
+    @property
+    def total_batch_idx(self):
+        return self.training_loop.total_batch_idx
+
+    @property
+    def batch_idx(self):
+        return self.training_loop.iteration_count
+
+    @property
+    def split_idx(self):
+        return self.training_loop.split_idx
+
+    @property
+    def min_steps(self):
+        return self.training_loop.min_steps
+
+    @property
+    def max_steps(self):
+        return self.training_loop.max_steps
+
+    @max_steps.setter
+    def max_steps(self, value):
+        # TODO(@awaelchli): This setter is required by debugging connector (fast dev run), should be avoided
+        self.training_loop.max_steps = value
+
+    @property
+    def running_loss(self):
+        return self.training_loop.batch_loop.running_loss
+
+    @property
+    def skip_backward(self) -> bool:
+        """ Determines whether the loop will skip backward during automatic optimization. """
+        return self.training_loop.batch_loop.skip_backward
+
+    @skip_backward.setter
+    def skip_backward(self, value: bool):
+        """ Determines whether the loop will skip backward during automatic optimization. """
+        self.training_loop.batch_loop.skip_backward = value
+
+    @property
+    def done(self) -> bool:
+        # TODO(@awaelchli): Move track steps inside training loop and move part of these condition inside training loop
+        stop_steps = self.max_steps is not None and self.global_step >= self.max_steps
+        stop_epochs = self.max_epochs is not None and self.current_epoch >= self.max_epochs
+
+        should_stop = False
+        if self.trainer.should_stop:
+            # early stopping
+            met_min_epochs = self.current_epoch >= self.min_epochs if self.min_epochs else True
+            met_min_steps = self.global_step >= self.min_steps if self.min_steps else True
+            if met_min_epochs and met_min_steps:
+                should_stop = True
+            else:
+                log.info(
+                    'Trainer was signaled to stop but required minimum epochs'
+                    f' ({self.min_epochs}) or minimum steps ({self.min_steps}) has'
+                    ' not been met. Training will continue...'
+                )
+                self.trainer.should_stop = False
+
+        return stop_steps or should_stop or stop_epochs
+
+    def connect(self, trainer: 'pl.Trainer', *args, **kwargs):
+        self.trainer = trainer
+        self.training_loop.connect(trainer)
+
+    def reset(self) -> None:
+        self.iteration_count = 0
+
+    def run(self):
+        if not self._should_skip_training():
+            return super().run()
+
+    def on_run_start(self):
```

Comment:
nit: please add typehints for below

will you add docs in a follow up PR?

Reply:
tracking this in #7938
@justusschock wants to do the typing <3